File: answers/Fab-Gaurav/Day 17/Answer_1.cpp
#include<bits/stdc++.h>
using namespace std;

void uniQuad(int arr[], int n, int k)
{
    int m, l, h;
    for(int i=0; i<=n-3; i++)
    {
        for(int j=i+1; j<=n-2; j++)
        {
            int m=k-(arr[i]+arr[j]);
            l=j+1;
            h=n-1;
            while(l<h)
            {
                if(arr[l] + arr[h] < m){
                    l++;
                }
                else if(arr[l] + arr[h] > m){
                    h--;
                }
                else{
                    cout<<"("<<arr[i]<<", "<<arr[j]<<", "<<arr[l]<<", "<<arr[h]<<")";
                    l ++;
                    h --;
                }
            }            
        }
    }
}

int main()
{
    int n, k, arr[20];
    cout<<"Enter the number of elements in array : ";
    cin>>n;
    cout<<"Enter the target sum : ";
    cin>>k;
    cout<<"Enter the elements of array : ";
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }

    cout<<"The unique quadruple is : ";
    uniQuad(arr, n, k);
    return 0;
}

File: answers/Shobhit403/DAY 17/Answer1.py
S=int(input('Enter the Size: '))
TSUM=int(input('Enter Target Sum: '))
a=[]
for i in range(S):
    a.append(int(input('Enter the array ElementS: ')))
def Quadruple(a,TSUM):
    a.sort()
    for i in range(S-3):
        for j in range(i+1,S-2):
            k=TSUM-(a[i]+a[j])
            l=j+1
            h=S-1
            while l<h:
                if a[l]+a[h]<k:
                    l+=1
                elif a[l]+a[h]>k:
                    h-=1
                else:
                    print("".join([str(a[i]),str(a[j]),str(a[l]),str(a[h])])+'$',end='')
                    l,h=l+1,h-1
Quadruple(a,TSUM)
